src/Elmish.WPF/ViewModel.fs

```diff
@@ -178,11 +179,11 @@ and internal BaseVmBinding<'model, 'msg> =
   | OneWaySeq of OneWaySeqBinding<'model, obj, obj, obj>
   | TwoWay of TwoWayBinding<'model>
   | Cmd of cmd: Command
-  | SubModel of SubModelBinding<'model, 'msg, obj, obj>
-  | SubModelWin of SubModelWinBinding<'model, 'msg, obj, obj>
-  | SubModelSeqUnkeyed of SubModelSeqUnkeyedBinding<'model, 'msg, obj, obj>
-  | SubModelSeqKeyed of SubModelSeqKeyedBinding<'model, 'msg, obj, obj, obj>
-  | SubModelSelectedItem of SubModelSelectedItemBinding<'model, 'msg, obj, obj, obj>
+  | SubModel of SubModelBinding<'model, 'msg, obj, obj, obj>
+  | SubModelWin of SubModelWinBinding<'model, 'msg, obj, obj, obj>
+  | SubModelSeqUnkeyed of SubModelSeqUnkeyedBinding<'model, 'msg, obj, obj, obj>
+  | SubModelSeqKeyed of SubModelSeqKeyedBinding<'model, 'msg, obj, obj, obj, obj>
+  | SubModelSelectedItem of SubModelSelectedItemBinding<'model, 'msg, obj, obj, obj, obj>
```

Comment:
It's fair to point out that these types are *always* using `obj` for their last several parameters, and therefore perhaps we could just be storing those fields directly as `obj` and not have those extra type parameters.

This is only true of the `...Binding<>` types, not of the `...Data<>` types (which have the box/unbox mapping)

Reply:
Yes, I agree.  IIRC, I added these type parameters even though they are always instantiated using `obj` so that I could make sense of the data contained in each case.